lms/djangoapps/courseware/tests/test_module_render.py

```diff
@@ -143,6 +144,22 @@ def progress(self, json_data, suffix):  # pylint: disable=unused-argument
         return self.runtime.publish(self, 'progress', {})
 
 
+class XBlockWithoutCompletionAPI(XBlock):
```

Comment:
Is there a test to cover the case of XBlocks that do emit the 'progress' signal and have `has_custom_completion = True`?  That is, something to cover the `handle_deprecated_progress_event()` function's `else` path that doesn't cause a `BlockCompletion` record to be created?

Reply:
Yes, there is one: [`test_progress_signal_ignored_for_completable_xblock`](https://github.com/open-craft/edx-platform/blob/2c70e38e7d9b92e6d5b374611d62d90cc7c61a50/lms/djangoapps/courseware/tests/test_module_render.py#L679).

Reply:
Ah, sorry, I must have been conflating two tests in my mind with my earlier comment.